EAP1300 fixes

ryancdotorg opened the pull request:
This re-adds the EnGenius EAP1300 to the firmware update script, and adds the EAP1300EXT as an alt model of the EAP1300.

Thanks to @cminyard for finishing the mac address fixes and DSA conversion.

ryancdotorg committed:
ipq40xx: eap1300: enable firmware upgrade

Signed-off-by: Ryan Castellucci <code@ryanc.org>

ryancdotorg committed:
ipq40xx: eap1300: add eap1300ext as alt model

The EnGenius EAP1300 and EAP1300EXT use an identical boards and firmware
(as flashed) from the vendor.

As with the EAP1300, the EAP1300EXT requires a specific firmware version
to flash OpenWRT. Unfortunately, the required firmware is truncated on
the vendor's website.

A working file can be created as follows:

```
curl \
https://www.engeniustech.com/wp_firmware/eap1300-all-v3.5.3.5_c1.9.04.bin \
| perl -pe 's/\x09EAP1300_A/\x0cEAP1300EXT_A/' \
> eap1300ext-all-v3.5.3.5_c1.9.04.bin
```

The file should have sha256:
`58a1197a426139a12b03fd432334e677124cbe3384349bd7337f2ee71f1dcfd4`.

Please see commit 2b4ac79 for further
details.

Vendor firmware may be flashed from OpenWRT by decrypting it. A tool
able to do that is available from:

https://github.com/ryancdotorg/enfringement/blob/main/decrypt.py

Signed-off-by: Ryan Castellucci <code@ryanc.org>

ryancdotorg commented:
These commits should also be pulled in against 23.05 as well, see also #15402

tmn505 commented on target/linux/ipq40xx/base-files/lib/upgrade/platform.sh:
```diff
@@ -115,6 +115,7 @@ platform_do_upgrade() {
 	cilab,meshpoint-one |\
 	edgecore,ecw5211 |\
 	edgecore,oap100 |\
+	engenius,eap1300 |\
```
This looks wrong. Device has NOR flash yet You try to use NAND upgrade procedure. Does the upgrade not work currently? Later in that file eap1300 is handled by asterisk which should suffice.

cminyard commented:
On Mon, May 06, 2024 at 12:19:40PM -0700, tmn505 wrote:
> @tmn505 commented on this pull request.
> 
> 
> 
> > @@ -115,6 +115,7 @@ platform_do_upgrade() {
>  	cilab,meshpoint-one |\
>  	edgecore,ecw5211 |\
>  	edgecore,oap100 |\
> +	engenius,eap1300 |\
> 
> This looks wrong. Device has NOR flash yet You try to use NAND upgrade procedure. Does the upgrade not work currently? Later in that file eap1300 is handled by asterisk which should suffice. 

This was where it originally was, I think.  I have done about 20
upgrades with this without issue.

ryancdotorg commented:
@cminyard @tmn505 I had added the device to platform.sh in my previous PR, but I've checked the branches and it was not there originally.

https://github.com/openwrt/openwrt/blob/openwrt-21.02/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

https://github.com/openwrt/openwrt/blob/openwrt-22.03/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

I do not remember why I added it before.

ryancdotorg commented:
I'm closing this, I will submit a PR that just adds the EAP1300EXT.
